keras/preprocessing/image.py

```diff
@@ -1109,6 +1117,20 @@ def __init__(self, x, y, image_data_generator,
                  data_format=None,
                  save_to_dir=None, save_prefix='', save_format='png',
                  subset=None):
+        if (type(x) is tuple) or (type(x) is list):
+            if type(x[1]) is not list:
+                x_misc = [np.asarray(x[1])]
+            else:
+                x_misc = [np.asarray(xx) for xx in x[1]]
+            x = x[0]
+            for xx in x_misc:
+                if len(x) != len(xx):
+                    raise ValueError('All of the arrays in `x` should have the same length. '
+                                     'Found a pair with: len(x[0]) = %s, len(x[?]) = %s' %
+                                     (len(x), len(xx)))
+        else:
+            x_misc = None
```

Comment:
Small comment, using x_misc as a list all the time would make your code much cleaner.
So I would replace x_misc = None with x_misc = []. It would remove a bunch of if.